### ISSUE TITLE
ttnn.addcmul: the composite fallback computes (b*c)*value while the LLK kernel and torch.addcmul compute (value*b)*c — the two paths disagree

### ISSUE THREAD

**Issue by botify-service** [tenstorrent/tt-metal#53282]
### Component / Area

ttnn eltwise ternary — `ttnn/cpp/ttnn/operations/eltwise/ternary/ternary_composite_op.cpp` (`_addcmul`), the composite fallback reached from `ttnn::addcmul` in `ternary.cpp`

### Issue Type (optional)

Bad Outputs

### Observed

`ttnn.addcmul` has two implementations that associate the multiplication differently. They
disagree with each other, and only one of them matches the registered golden.

**The LLK path is correct.** `ckernel_sfpu_addcmul.h` multiplies the scalar into `input_b`
before touching `input_c`:

```cpp
sfpi::vFloat a_prod = value_float * a_in1;        // value * b
sfpi::vFloat a_res  = a_prod * a_in2 + a_in0;     // (value * b) * c + a
```

**The composite fallback is not.** `_addcmul` forms the tensor product first and applies the
scalar last:

```cpp
// ternary_composite_op.cpp:27-31
Tensor t_mul    = ttnn::multiply(input_b, input_c, ...);   // b * c
Tensor t_factor = ttnn::multiply(t_mul, value, ...);       // (b * c) * value
Tensor result   = ttnn::add(input_a, t_factor, ...);
```

The registered golden is `torch.addcmul`, whose kernel is
`self_val + scalar_val * t1_val * t2_val` — left-to-right, so `(value * t1) * t2`
(`aten/src/ATen/native/cpu/PointwiseOpsKernel.cpp`, both the reduced-float and the general
branch). That is the LLK association, not the fallback's.

Multiplication is associative in exact arithmetic and not in floating point. A small `value`
exists precisely to scale the product down; applying it last means the unscaled `b * c` has to
be representable first.

### Cases

`float32`, `a = 0` unless noted. Every exact result below is an ordinary `float32`:

| a | b | c | value | fallback `(b*c)*value` | `torch.addcmul` |
|---|---|---|---|---|---|
| 0 | 1e30 | 1e30 | 1e-30 | **inf** | 1e30 |
| 0 | 1e25 | 1e25 | 1e-25 | **inf** | 1e25 |
| 0 | 3e19 | 3e19 | 1e-20 | **inf** | 9e18 |
| 1 | 1e35 | 1e35 | 1e-40 | **inf** | 9.99995e29 |
| 0 | 1e-30 | 1e-30 | 1e30 | **0** | 1e-30 |
| 0 | 2 | 3 | 4 | 24 | 24 (ok — control) |

Both directions fail: `b * c` overflows to `inf` when the scale-down comes last, and underflows
to `0` when a large `value` was meant to scale a tiny product back up. The control row shares the
run, so this is not general breakage.

### Where the fallback is reached

`ternary.cpp` routes to `_addcmul` rather than the LLK op when:

```cpp
if (is_invalid_bcast(broadcast_type) || (is_any_input_block_format && is_subtile_bcast)) {
    log_debug(tt::LogOp, "Addcmul Fallback - TTT");
    return _addcmul(input_a, input_b, input_c, value_f, memory_config);
}
```

so the same call with the same numbers can take either association depending on the broadcast
shape and whether an input is block-float. That is the part I would flag as the actual defect
even setting the range question aside: one op, two answers.

### Expected

`ttnn.addcmul` should associate as `(value * input_b) * input_c` on every path, matching both the
LLK kernel and the registered golden. Concretely, in `_addcmul`:

```cpp
-    Tensor t_mul    = ttnn::multiply(input_b, input_c, ...);
-    Tensor t_factor = ttnn::multiply(t_mul, value, ...);
+    Tensor t_scaled = ttnn::multiply(input_b, value, ...);
+    Tensor t_factor = ttnn::multiply(t_scaled, input_c, ...);
     Tensor result   = ttnn::add(input_a, t_factor, ...);
```

That is a two-line reordering with no extra op and no extra intermediate — the same three
element-wise calls, in the order the reference uses. It removes the path disagreement and the
range loss together.

`_addcdiv` in the same file is already consistent: it computes `(value * b) / c`, matching
`scalar_val * t1_val / t2_val`. No change needed there — I checked it while I was in the file.

### Steps to reproduce

I do not have a Tenstorrent card. These are the commands for someone who does. The inputs must
take the fallback path — a broadcast shape the LLK op does not handle, or a block-float dtype
with a subtile broadcast — otherwise the correct LLK association is used and the test passes.

```python
import torch
import ttnn

device = ttnn.open_device(device_id=0)

cases = [(0.0, 1e30, 1e30, 1e-30), (0.0, 3e19, 3e19, 1e-20),
         (1.0, 1e35, 1e35, 1e-40), (0.0, 1e-30, 1e-30, 1e30),
         (0.0, 2.0, 3.0, 4.0)]

print(f"{'b':>10} {'c':>10} {'value':>10} {'ttnn':>14} {'torch':>14}  ok")
for a, b, c, v in cases:
    ta, tb, tc = (ttnn.from_torch(torch.full((1, 1, 32, 32), x, dtype=torch.float32),
                  dtype=ttnn.float32, layout=ttnn.TILE_LAYOUT, device=device)
                  for x in (a, b, c))
    got = ttnn.to_torch(ttnn.from_device(ttnn.addcmul(ta, tb, tc, value=v)))[0, 0, 0, 0].item()
    want = torch.addcmul(torch.tensor(a), torch.tensor(b), torch.tensor(c), value=v).item()
    print(f"{b:>10.3g} {c:>10.3g} {v:>10.3g} {got:>14.6g} {want:>14.6g}  "
          f"{'ok' if got == want else 'FAIL'}")

ttnn.close_device(device)
```

A cleaner oracle that needs no magnitudes at all: run the same inputs twice, once on a shape that
takes the LLK path and once on a shape that takes the fallback, and assert the two agree. Today
they do not.

### Why the current tests do not catch it

- The ternary sweeps draw `input_b` and `input_c` from moderate ranges and leave `value` at or
  near `1`, so the product never approaches the representable edge and the association cannot
  show.
- No test compares the LLK path against the fallback path for the same inputs, which is the
  check that would catch this independently of the values sampled.
- #33759 ("Validate addcmul accuracy", closed) validated accuracy but not the fallback's
  ordering against the LLK op.

### Is this a regression?

No — I did not bisect, and I have no evidence of a commit that changed the ordering. Context: the
LLK kernel carries a 2026 copyright and the composite a 2024 one, so the likely history is that
the newer kernel was written to the reference's association while the older fallback kept its own.
Adjacent but different: #48240 item 5 concerns `scalar_input_a`/`scalar_input_b` on the
ADDCMUL/ADDCDIV scalar variant and is marked NOT A BUG; that is a plumbing question, this is the
arithmetic order.

### Software Versions

| | |
|---|---|
| tt-metal | `main`, source as of 2026-08-16 |
| reference | `torch.addcmul` per the registered golden in `ttnn/ttnn/operations/ternary.py` |

### Hardware Details

No Tenstorrent card and no ttsim run — stating that up front.

The association orders are read from the source: `_addcmul` in `ternary_composite_op.cpp`, the
SFPU kernel in `ckernel_sfpu_addcmul.h`, the dispatch in `ternary.cpp`, and PyTorch's
`PointwiseOpsKernel.cpp`. The table is both formulas evaluated in `float32` with every
intermediate rounded to `float32` — the same three element-wise operations the composite issues,
in the same order. That models host-side composition of `float32` element-wise ops, which is what
`_addcmul` builds; it does not model SFPU rounding.

The claim I am most confident in needs no numbers at all and no hardware: the two paths associate
differently, and only one matches the golden. The specific `inf`/`0` values follow from `float32`
and should reproduce, but a run would confirm which shapes actually land on the fallback.

### Impact

`addcmul` is the fused multiply-accumulate behind optimizer updates — Adam's second-moment term is
`addcmul(avg_sq, grad, grad, value=1-beta2)`, where a small `value` multiplying a squared gradient
is exactly the shape that loses range here. The failure is silent: `inf` propagates into the
optimizer state and surfaces as `NaN` later, and the `0` case quietly stalls an update. It is
gated behind the fallback, so it will hit some shapes and not others, which makes it harder to
attribute when it does bite.
